java-checks/src/main/java/org/sonar/java/checks/ExceptionsShouldBeImmutableCheck.java

```diff
@@ -66,8 +70,8 @@ private static boolean isFinal(VariableTree member) {
   }
 
   private static boolean isException(ClassTree classTree) {
-    IdentifierTree simpleName = classTree.simpleName();
-    return simpleName != null && (simpleName.name().endsWith("Exception") || simpleName.name().endsWith("Error"));
+    Type type = classTree.symbol().type();
+    return type.isSubtypeOf("java.lang.Exception") || type.isSubtypeOf("java.lang.Error");
```

Comment:
wondering if we shouldn't check for any throwable maybe ? or test that it is a subtype of throwable but not throwable itself ?